cupyx/scipy/interpolate/_rbfinterp.py

```diff
@@ -0,0 +1,738 @@
+"""Module for RBF interpolation."""
+import warnings
+from itertools import combinations_with_replacement
+
+import cupy as cp
+from scipy.spatial import KDTree        # FIXME
+from scipy.special import comb          # FIXME
+
+
+# individual kernels et al
+
+def linear(r):
+    return -r
+
+
+def thin_plate_spline(r):
+    if r == 0:
+        return 0.0
+    else:
+        return r**2*cp.log(r)
```

Comment:
All these small functions will result in several kernel invocations, they could be combined into a single one for performance gains :)

There is an example on how to achieve this here

https://github.com/cupy/cupy/blob/master/cupyx/scipy/special/_stats_distributions.py#L37-L88

Reply:
Great, thanks for the pointer! This is insanely cool. 
It does indeed make a big difference, roughly halving the time to run the tests.

I've to admit I'm somewhat confused by the invocation, especially the what goes into which tuple arguments:

```
log_ndtr = _core.create_ufunc(
    'cupyx_scipy_special_log_ndtr',
    (('f->f', 'out0 = log_ndtrf(in0)'), 'd->d'),     # not (('f->f', 'signature'),  ('d->d', signature)) ?
    'out0 = log_ndtr(in0)',                          # this tacitly assumes the same name as the ufunc?
```

Sure the example is not hard to follow when pointed at, so this comment is a gentle request for documenting the syntax :-).

Reply:
<s>halving the time to run the tests</s> Scratch that. Moving python loops to ufuncs + broadcasting makes the test run locally from 70s to 2s.  In fairness, the scipy version had loops in pythranized C++, so lifting them back to pure python was bound to be slow. So the this PR is all ufuncs now.